src/test/controller/BaseController.ts

```diff
@@ -1,21 +1,57 @@
 import express, { Express } from 'express';
 import { App } from '../../main/App';
 import redisMock, { RedisClient } from 'redis-mock';
+import { auth } from 'firebase-admin/lib/auth';
+import Auth = auth.Auth;
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const firebaseMock = require('firebase-mock');
 
 /**
  * Creates an express app and registers all handlers and routes
  * * Used for integration tests that test controllers (supertest needs an instance of express app without the port binding)
  * */
 export const registerTestApplication = async (
-  redisClient: RedisClient
+  redisClient: RedisClient,
+  firebaseAdmin: Auth
 ): Promise<Express> => {
   const app = new App();
   const expressApp = express();
-  await app.registerHandlersAndRoutes(expressApp, redisClient);
+  await app.registerHandlersAndRoutes(expressApp, redisClient, firebaseAdmin);
   return expressApp;
 };
 
 export const connectRedisClient = async (): Promise<RedisClient> => {
   // see https://www.npmjs.com/package/redis-mock
   return redisMock.createClient();
 };
+
+export const createFirebaseMock = (): Auth => {
+  const mockAuth = new firebaseMock.MockAuthentication();
+  const mockDatabase = new firebaseMock.MockFirebase();
+  const mockFirestore = new firebaseMock.MockFirestore();
+  const mockStorage = new firebaseMock.MockStorage();
+  const mockMessaging = new firebaseMock.MockMessaging();
+  const mockSdk = new firebaseMock.MockFirebaseSdk(
+    // use null if your code does not use RTDB
```

Comment:
Not sure if we aren't using any of these, but if you wanted to keep it for developers, maybe summarize all these "use null..." comments into one?

Reply:
This is the part I'm not really understanding, I just followed the way of setting up of firebase mock based on the document.

Reply:
Do you mind pasting the link? I could take a look! :)

Reply:
This is the link for firebase mock library.
https://github.com/dmurvihill/firebase-mock

Reply:
@yunishimura0716 do you think this one would work https://github.com/mindhivefi/ts-mock-firebase

Reply:
@ericrkuo The error occurred when I add `ts-mock-fireabse` because of build error for `gyp` and `node-pre-gyp` so that I could not install the package properly.

Reply:
Do you have the build output?

Reply:
This is the error thrown when I tried to install `ts-mock-firebase`
![Screen Shot 2021-03-02 at 12 01 58 AM](https://user-images.githubusercontent.com/38684796/109616997-b0ad1080-7aea-11eb-8c91-fa29c471375d.png)

Reply:
Ah I see, okay let's forget about that mock library. I'm okay w using this current mocking library